### ISSUE TITLE
nksr sample ziwei

### ISSUE THREAD

**Issue by ZBoIsHere** [ZBoIsHere/ZBoIsHere#15]
import argparse
import torch
import numpy as np
from nksr import Reconstructor, utils, fields
from pycg import vis
from pathlib import Path

import ssl
import trimesh

if __name__ == '__main__':
    parser = argparse.ArgumentParser(description='Parse nksr reconstruct args.')
    parser.add_argument('--data', type=str, default='/app/data/sample.ply', help='path of data', required=True)
    parser.add_argument('--output', type=str, default='/app/data/sample_output.obj', help='path of output data', required=True)
    args = parser.parse_args()

    device = torch.device("cuda:0")
    # 此处修改为输入点云文件名
    geom = vis.from_file(args.data)

    # 提取点坐标，颜色，法线信息
    input_xyz = torch.from_numpy(np.asarray(geom.points)).float().to(device)
    input_normal = torch.from_numpy(np.asarray(geom.normals)).float().to(device)
    input_color = torch.from_numpy(np.asarray(geom.colors)).float().to(device)

    # 创建nksr对象
    reconstructor = Reconstructor(device)
    # detail_level为可调节参数，取值范围0~1，表示mesh的精细程度，精细程度越高，mesh质量越好，但面片数变多
    field = reconstructor.reconstruct(input_xyz, input_normal, detail_level=0.8)
    field.set_texture_field(fields.PCNNField(input_xyz, input_color))
    mesh = field.extract_dual_mesh(mise_iter=1)

    mesh_np = trimesh.Trimesh(vertices=mesh.v.detach().cpu().numpy(), faces=mesh.f.detach().cpu().numpy(),
                              vertex_colors=mesh.c.detach().cpu().numpy(), process=False)
    # 此处修改为输出mesh文件路径
    mesh_np.export(args.output)
